src/frame/sdk/GroupApi.ts

```diff
@@ -114,27 +105,43 @@ function getCompleteListOfUserPublicKeys(userList: string[]) {
     });
 }
 
+/**
+ * Retrieves the master public keys for users in the memberList, adminList and owner.
+ */
+function validateUsers(memberList: string[], adminList: string[], maybeOwnerUserId?: string) {
+    const ownerUserId = maybeOwnerUserId ? [maybeOwnerUserId] : [];
+    const uniqueUsers = Array.from(new Set([...memberList, ...adminList, ...ownerUserId]).values());
+    return getCompleteListOfUserPublicKeys(uniqueUsers).map((userList) => {
+        return {
+            memberKeys: userList.filter((user) => memberList.includes(user.id)),
+            adminKeys: userList.filter((user) => adminList.includes(user.id) || ownerUserId.includes(user.id)),
+        };
+    });
+}
+
 /**
  * Create a new group with the provided ID and options
- * @param {string}  groupID Client provided optional group ID
- * @param {string}  groupName Client provide optonal group name
- * @param {boolean} addAsMember Whether the group creator shoudl be added as a member upon create
- * @param {boolean} needsRotation Whether the group needs to rotate it private key
+ * @param groupID Client provided optional group ID
+ * @param groupName Client provide optonal group name
+ * @param needsRotation Whether the group needs to rotate it private key
+ * @param memberList Client provide optonal user list to be initialized as group members
+ * @param adminList List of atlease one user to be initialized as group administrators. AdminList will contain the group owner or creator if an owner is not provided
+ * @param ownerUserId Client provide user designated to be the group owner, if undefined the group creator will be the owner by defalt on the server side
  */
 export function create(
     groupID: string,
     groupName: string,
-    addAsMember: boolean,
     needsRotation: boolean,
-    memberList?: string[]
+    memberList: string[],
+    adminList: string[],
+    ownerUserId?: string
 ): Future<SDKError, GroupDetailResponse> {
-    const creatorMemberKey = addAsMember ? [{id: ApiState.user().id, masterPublicKey: publicKeyToBase64(ApiState.userPublicKey())}] : [];
-    const memberListValidationResult = memberList !== undefined && memberList.length > 0 ? getCompleteListOfUserPublicKeys(memberList) : Future.of([]);
-
-    return memberListValidationResult
-        .flatMap((memberKeys) => GroupOperations.groupCreate(ApiState.userPublicKey(), ApiState.signingKeys(), creatorMemberKey.concat(memberKeys)))
-        .flatMap(({encryptedGroupKey, groupPublicKey, transformKeyGrantList}) =>
-            GroupApiEndpoints.callGroupCreateApi(groupID, groupPublicKey, encryptedGroupKey, needsRotation, transformKeyGrantList, groupName)
+    return validateUsers(memberList, adminList, ownerUserId)
+        .flatMap(({memberKeys, adminKeys}) => {
+            return GroupOperations.groupCreate(ApiState.signingKeys(), memberKeys, adminKeys);
```

Comment:
Lets be consistent here like you are below on line 143 and remove the braces and the `return` and just have `.flatMap(({memberKeys, adminKeys}) => GroupOperations.groupCreate...`